lib.php

```diff
@@ -60,6 +60,14 @@ public function get_form_elements_module($mform, $context, $modulename = "") {
 
         $location = ($context == context_system::instance()) ? 'defaults' : 'module';
 
+        // Get whether plugin is enabled for this module.
+        $moduletiienabled = empty($modulename) ? "0" : get_config('plagiarism_turnitinsim', 'turnitinmodenabled'.substr($modulename, 4));
```

Comment:
`turnitinmodenabled'.substr($modulename, 4)`

Does this work to get the right config values such as turnitinmodenabledassign, turnitinmodenabledforum and turnitinmodenabledworkshop? I'm a bit confused by the substr($modulename, 4) bit, wouldn't that mean it only checks for turnitinmodenabledassi for example?

Reply:
`$modulename` contains values in below format

- mod_assign
- mod_forum
- mod_workshop

Reply:
That makes sense, good way to do it!